frontend/src/components/collab/chat/ChatSpace.tsx

```diff
@@ -0,0 +1,119 @@
+"use client";
+import { useEffect, useRef, useState } from "react";
+import ChatBubble from "./ChatBubble";
+import { BsSendFill } from "react-icons/bs";
+import { RxCross2 } from "react-icons/rx";
+import { Button, Divider } from "@nextui-org/react";
+import ProfilePictureAvatar from "@/components/common/ProfilePictureAvatar";
+import { useCollabContext } from "@/contexts/collab";
+import ChatMessage from "@/types/chat_message";
+import { v4 as uuid } from "uuid";
+
+interface IChatSpaceProps {
+  onClose: () => void;
+}
+
+const ChatSpace = ({ onClose }: IChatSpaceProps) => {
+  const { partner, user, socketService } = useCollabContext();
+
+  if (!socketService || !partner || !user) return null;
```

Comment:
what happens if one of them is null? will the chat just not render?

Reply:
Yep, but the existence of this entire page assumes that a valid matching and socket service has been established, so  by right none of these should be null.. This is required because otherwise the typing will be invalid... not sure if this is the correct way to overcome it 😢

Reply:
I never pull your branch locally as I am tracking js branch. Is it correct to say that if one of these thing is null (service not found, partner not found or user not login), then the chat button will appear, but clicking that button will not open the chat space?

Reply:
Yep, but this is the flow of rendering: 
1. If own user does not exist, auth middleware will not allow routing to the collaboration page. 
2. If user exists, collaboration page mounts, calls `handleConnectToRoom()` in the `CollabProvider` (frontend/src/contexts/collab.tsx). This function will handle setting all the match configs (language, partner, etc) and setting up of the socket service. While all this is being done, `isLoading` is set to True. Only when everything is done and there is no error, then `isLoading` is set to False.
3. While `isLoading` is true, the page will render only the loading component. So if the child components are visible, it means that all the configurations are in place, and it is not possible for any of those values to be null.

The only thing i'm concerned about is error handling in the `handleConnectToRoom()` function, as it throws `notFound()`, which would need to be handled gracefully by the frontend. I didn't alter much of the original code in this function, and I think @tryyang2001 you wrote it, so may I know if you had a plan for this?